src/all.js

```diff
@@ -31,6 +32,11 @@ function initAll () {
   nodeListForEach($radios, function ($radio) {
     new Radios($radio).init()
   })
+
+  var $tabs = document.querySelectorAll('[data-module="tabs"]')
```

Comment:
At the moment the macro template has the data attribute on the container. This means that if I use the macro, it will automatically get instantiated into tabs.

This could become problematic if I want to do something more custom. Like listen to an event fired by the instance (I know we can't do that right now).

Is GOV.UK Frontend continuing to instantiate automatically like this?

Reply:
You concern is valid, but this is consistent with how the other components are being initialised. Will make a note and look into that for all the components.

Reply:
Yep, it might be okay, broadly speaking, for say tabs and conditional radio button reveals. But as a general approach it can be overly constraining.

Thanks @alex-ju

Reply:
This use case is already accounted for: you can avoid `initAll()` and call the constructor yourself.

Reply:
Since we did work to stop `initAll()` from executing automatically :)

Reply:
Yes, not initialising or initialising separately a specific component is also an option.
When Adam mentioned 'something custom', I was thinking of a case when you want to initialise but hook different listeners or add custom functionality to a Tabs instance before doing that, which I think is still possible with the current model:
```
var myTabs = new Tabs($tabs)
myTabs.myCustomFunction = function () { ... }
myTabs.init()
```